common/cloud_builder.py

```diff
@@ -210,14 +214,29 @@ def _cancel_build(self, build_id: str) -> None:
     """Cancel a GCB build"""
     self.builds.cancel(projectId=self.project_id, id=build_id).execute()
 
+  def _extract_chat_history(self, full_log: str) -> str:
+    """Extracts the agent chat history from cloud build log."""
+    in_chat = False
+    pattern = r'^Step\s+#(\d+)\s+-\s+"agent-step":\s+'
```

Comment:
nit: can you put this in `_CHAT_HISTORY_PATTERN = re.compile(...)` somewhere?

Reply:
Done, thanks